internal/zinc-persist/src/main/scala/sbt/internal/inc/RelationsTextFormat.scala

```diff
@@ -0,0 +1,128 @@
+package sbt.internal.inc
+
+import java.io.{ File, BufferedReader, Writer }
+
+import sbt.internal.util.Relation
+import xsbti.api.DependencyContext._
+
+trait RelationsTextFormat extends FormatCommons {
+
+  def mappers: AnalysisMappers
+
+  private case class Descriptor[A, B](
+    header: String,
+    selectCorresponding: Relations => Relation[A, B],
+    keyMapper: Mapper[A],
+    valueMapper: Mapper[B]
+  )
+
+  private def stringsDescriptor(header: String, rels: Relations => Relation[String, String]) =
+    Descriptor(header, rels, Mapper.forString, Mapper.forString)
+
+  private val allRelations: List[Descriptor[_, _]] = {
```

Comment:
Agree on documentation part but IMO first we need to clean up whole logic around relation construction. Code (from below):

```scala
relations match {
      case p :: bin :: lcn :: mri :: mre :: ii :: ie :: lii :: lie :: cn :: un :: bcn :: Nil =>
```

Is IMO one of the most hacky thing in whole zinc codebase and at some point it needs to be cleanup and refactored.
I tried to do so in this PR but after some time I look at diff and I was way to big for single PR (and I needed to fit cached compilation related changes).